back/app/services/survey_results_generator_service.rb

```diff
@@ -24,23 +24,31 @@ def generate_results
   end
 
   def visit_select(field)
+    option_keys = field.options.each_with_object({}) do |option, accu|
+      accu[option.key] = option.key
+    end
     values = inputs
       .select("custom_field_values->'#{field.key}' as value")
       .where("custom_field_values->'#{field.key}' IS NOT NULL")
     distribution = Idea.select(:value).from(values).group(:value).order(Arel.sql('COUNT(value) DESC')).count.to_a
+    filtered_distribution = distribution.select { |(value, _count)| option_keys.value? value }
```

Comment:
Following on from the simplification: 
```suggestion
     filtered_distribution = distribution.select { |(value, _count)| option_keys.include? value }
```